src/net/listener.rs

```diff
@@ -321,6 +322,7 @@ mod test {
             })
             .join({
                 socket_incoming
+                .map_err(|e| panic!("incoming error: {}", e))
```

Comment:
I guess you accidentally left `panic!`?

Reply:
Nah, it's in a test, it's supposed to panic.

Reply:
right, didn't check context :)